synapseclient/client.py

```diff
@@ -1132,6 +1132,47 @@ def _getEntityBundle(self, entity, version=None, bitFlags=0x800 | 0x400 | 0x2 |
 
         return bundle
 
+    def copy(self, obj, parentId):
+        """
+        Copies most recent version of a file to a specified synapse ID.
+
+        :param obj: A synapse ID or entity of a file
+                    
+        :param parentId: Synapse ID of a folder/project that the file wants to be copied to
+       
+        """
+        ent = self.get(obj, downloadFile=False)
+        profile = self.getUserProfile().ownerId
+        store = True
+        path = ent.path
+        #CHECK: Must be a file entity
+        if ent.entityType!='org.sagebionetworks.repo.model.FileEntity':
+            raise ValueError('"synapse cp" can only copy files!')
+        #Grab file handle createdBy annotation to see the user that created fileHandle
+        createdBy = self.restGET('/entity/%s/filehandles'%ent.id)['list'][0]['createdBy']
```

Comment:
One of these will be guaranteed to find the right file handle in case there's more than one:

syn._getFileHandle(ent.dataFileHandleId)

handles = self.restGET('/entity/%s/filehandles'%ent.id)['list']
for handle in handles:
  if handle['id'] == ent.dataFileHandleId:
     ...